### PR TITLE
UJS: Do not disable previously disabled elements

### PR DESCRIPTION
**Pull request by WoH**
### Description

This PR aims to fix a corner case concerning UJS behavior.
Reproduction steps are laid out in #33889.
The issue occurs because the a[data-disable(-with)] within a form will cause both the

` delegate document, Rails.formSubmitSelector, 'ajax:send', disableElement`

 which disables all child form elements as well as the 

`delegate document, Rails.linkClickSelector, 'click', disableElement`

to attempt to disable the element.
The first invocation change the text inside the a tag to be saved as attribute while being replaced by the data-disable-with attribute data.
The second invocation again takes the inner text (now the data-diable-with) and overwrites the saved text making the link always show the data-disabled-with text, even after reenabling, since the original text, saved within an attribute was overwritten during the second invocation.

Tl;Dr: Disabling an element that already is disabled is not handled well.

### Fixes

While you could make sure none of the handlers target the same elements, disabling a disabled element should not cause issues at any time. Since the current implementation already marks an element as disabled by ujs, we should use that information to determine if we should actually disable or just return early.

### CODE REVIEW

**Comment by WoH** [comment]
@javan may I ask you to take a look at this?

**Comment by javan** [comment]
lgtm!

**Comment by thelucid** [comment]
I am finding that the text from `disable_with` is not reverting to the `a` text for a successful Ajax response e.g. `disable_with: 'Working...'` is remaining as `Working...` regardless of outcome.

**Comment by WoH** [comment]
If you could open an issue with reproduction steps feel free to tag me and I'll take a look :+1:

**Comment by thelucid** [comment]
Apologies, my project was locked on an older version of rails-ujs. Works as expected.
